### DISCUSSION TITLE
Supporting Configuration in JSON format

### DISCUSSION THREAD

**Discussion by KhaledNabli** [rtl-airband/RTLSDR-Airband#214] [Ideas]
Hi There,

I'm managing somehow complicated config files (3 devices, 27 channels, 6 mixers). The application is performing well!
Unfortunately editing the the config files via ssh on nano / vim is no fun.

Since the syntax of the config file seems close the the JSON format, I started writing a web ui to generate the config files based on some user input (mainly frequencies). I can make that available online if there is interest in this.

Here comes the question, would it be worth it to make the application accept the configuration in JSON format directly without the hassle of replacing [] to () etc..

Thanks and a happy new year!
Khaled

**Comment by szpajder** [comment]
JSON is a data interchange format. It's not a configuration file format. It does not allow comments and the amount of required punctuation is in fact larger than in libconfig format (for example, option names are JSON keys, so they must be quoted, while in libconfig they don't). Feel free to use JSON for internal purposes of your webapp. However I don't see a reason to change the config format of rtl_airband - to JSON, in particular.
